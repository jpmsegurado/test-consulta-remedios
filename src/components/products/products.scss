.products {
  width: 100%;
  display: flex;
  flex-wrap: wrap;

  > .product-wrapper {
    width: calc(100% / 3);
    padding: 15px;
    box-sizing: border-box;
  }
}