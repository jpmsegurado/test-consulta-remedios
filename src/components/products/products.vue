<template>
  <div class="products">
    <div class="product-wrapper" v-for="product in products" :key="product.id">
      <product :product-info="product" />
    </div>
  </div>
</template>

<script>
import product from "../product/product";
import { mapGetters, mapActions } from "vuex";

export default {
  components: {
    product
  },
  computed: {
    ...mapGetters({
      products: "PRODUCT/ALL"
    })
  },
  methods: {
    ...mapActions({
      loadProducts: "PRODUCT/LOAD_PRODUCTS"
    })
  },
  mounted() {
    this.loadProducts();
  }
};
</script>

<style lang="scss" scoped src="./products.scss" />
