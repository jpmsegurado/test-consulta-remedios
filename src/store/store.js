import Vue from "vue";
import Vuex from "vuex";
import PRODUCT from "./modules/products";
import CART from "./modules/cart";

Vue.use(Vuex);

export default new Vuex.Store({
  modules: {
    PRODUCT,
    CART
  }
});
