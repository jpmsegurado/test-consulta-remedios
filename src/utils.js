export function formatMoney(value, decimal = 2) {
  const formatter = new Intl.NumberFormat("pt-BR", {
    style: "currency",
    currency: "BRL",
    minimumFractionDigits: decimal
  });

  return formatter.format(value);
}
