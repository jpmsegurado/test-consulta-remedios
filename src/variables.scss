$product-bg: #EEEEEE;
$product-name: #746A6A;
$product-price: #3486E6;
$cart-border: #E1E1E1;
$cart-text: #746A6A;
$item-cart-name: #7F7575;
$white: #fff;
$cart-button: #54A3FF;
$select-border: #E1E1E1;

.fade-enter-active, .fade-leave-active {
  transition: opacity .3s;
}
.fade-enter, .fade-leave-to {
  opacity: 0;
}