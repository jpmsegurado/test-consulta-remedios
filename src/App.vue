<template>
  <div id="app">
    <div class="row">
      <div class="products-wrapper">
        <h2>Games</h2>
        <mySelect v-model="selected" :value="selected" :list="options" />
        <products :selectedOption="options[selected]" />
      </div>

      <div class="shopping-cart-wrapper">
        <shopping-cart />
      </div>
    </div>
  </div>
</template>

<script>
import products from "./components/products/products";
import shoppingCart from "./components/shopping-cart/shopping-cart";
import mySelect from "./components/select/select";

export default {
  name: "app",
  components: {
    products,
    shoppingCart,
    mySelect
  },
  data() {
    return {
      options: [
        { label: "Mais populares", prop: "score" },
        { label: "Mais baratos", prop: "price" },
        { label: "Ordem Alfab√©tica", prop: "name" }
      ],
      selected: 0
    };
  }
};
</script>

<style lang="scss">
@import url("https://fonts.googleapis.com/css?family=Open+Sans");
body {
  padding: 0;
  margin: 0;
}
#app {
  font-family: "Open Sans";
  padding: 42px 70px;

  > .row {
    display: flex;
    width: 100%;
    flex-wrap: wrap;

    .products-wrapper {
      width: 75%;

      > h2 {
        font-size: 48px;
        margin-top: 0;
        margin-bottom: 44px;
        width: calc(100% - 277px);
        display: inline-block;
      }

      > .select {
        width: 262px;
        margin-right: 15px;
      }
    }

    .shopping-cart-wrapper {
      width: 25%;
    }
  }
}
</style>
