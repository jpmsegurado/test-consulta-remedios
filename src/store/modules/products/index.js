import products from "../../../assets/products.json";

const state = {
  products: []
};

const getters = {
  ALL: state => state.products
};

const actions = {
  LOAD_PRODUCTS: ({ commit }) => {
    return Promise.resolve(commit("SET_PRODUCTS", products));
  }
};

const mutations = {
  SET_PRODUCTS: (state, products) => {
    state.products = products;
  }
};

export default {
  namespaced: true,
  state,
  getters,
  actions,
  mutations
};
