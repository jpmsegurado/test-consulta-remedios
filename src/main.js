import Vue from "vue";
import App from "./App.vue";
import store from "./store/store";
import { formatMoney } from "./utils";

Vue.config.productionTip = false;

Vue.prototype.formatMoney = formatMoney;

new Vue({
  store,
  render: h => h(App)
}).$mount("#app");
