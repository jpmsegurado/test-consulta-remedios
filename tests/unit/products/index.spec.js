import { mount } from "@vue/test-utils";
import products from "@/components/products/products.vue";
import store from "@/store/store";
import { formatMoney } from "../../../src/utils";

const options = [
  { label: "Mais populares", prop: "score" },
  { label: "Mais baratos", prop: "price" },
  { label: "Ordem AlfabÃ©tica", prop: "name" }
];

describe("products.vue", () => {
  it("render products list normally", () => {
    const wrapper = mount(products, {
      store,
      mocks: {
        formatMoney
      }
    });
    wrapper.vm.$nextTick(() => {
      expect(wrapper).toMatchSnapshot();
    }, 1000);
  });

  it("render products list ordered by score", () => {
    const wrapper = mount(products, {
      store,
      mocks: {
        formatMoney
      },
      propsData: {
        selectedOption: options[0]
      }
    });
    wrapper.vm.$nextTick(() => {
      expect(wrapper).toMatchSnapshot();
    }, 1000);
  });

  it("render products list ordered by price", () => {
    const wrapper = mount(products, {
      store,
      mocks: {
        formatMoney
      },
      propsData: {
        selectedOption: options[1]
      }
    });
    wrapper.vm.$nextTick(() => {
      expect(wrapper).toMatchSnapshot();
    }, 1000);
  });

  it("render products list ordered by name", () => {
    const wrapper = mount(products, {
      store,
      mocks: {
        formatMoney
      },
      propsData: {
        selectedOption: options[2]
      }
    });
    wrapper.vm.$nextTick(() => {
      expect(wrapper).toMatchSnapshot();
    }, 1000);
  });
});
