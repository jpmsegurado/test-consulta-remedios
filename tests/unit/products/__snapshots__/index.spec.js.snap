// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`products.vue render products list normally 1`] = `
<div class="products">
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Super Mario Odyssey</p>
        <p>R$&nbsp;197.88</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Call Of Duty Infinite Warfare</p>
        <p>R$&nbsp;49.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>The Witcher III Wild Hunt</p>
        <p>R$&nbsp;119.50</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Call Of Duty WWII</p>
        <p>R$&nbsp;249.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Mortal Kombat XL</p>
        <p>R$&nbsp;69.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Shards of Darkness</p>
        <p>R$&nbsp;71.94</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Terra Média: Sombras de Mordor</p>
        <p>R$&nbsp;79.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>FIFA 18</p>
        <p>R$&nbsp;195.39</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Horizon Zero Dawn</p>
        <p>R$&nbsp;115.80</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
</div>
`;

exports[`products.vue render products list ordered by name 1`] = `
<div class="products">
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Call Of Duty Infinite Warfare</p>
        <p>R$&nbsp;49.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Call Of Duty WWII</p>
        <p>R$&nbsp;249.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>FIFA 18</p>
        <p>R$&nbsp;195.39</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Horizon Zero Dawn</p>
        <p>R$&nbsp;115.80</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Mortal Kombat XL</p>
        <p>R$&nbsp;69.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Shards of Darkness</p>
        <p>R$&nbsp;71.94</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Super Mario Odyssey</p>
        <p>R$&nbsp;197.88</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Terra Média: Sombras de Mordor</p>
        <p>R$&nbsp;79.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>The Witcher III Wild Hunt</p>
        <p>R$&nbsp;119.50</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
</div>
`;

exports[`products.vue render products list ordered by price 1`] = `
<div class="products">
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Call Of Duty Infinite Warfare</p>
        <p>R$&nbsp;49.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Mortal Kombat XL</p>
        <p>R$&nbsp;69.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Shards of Darkness</p>
        <p>R$&nbsp;71.94</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Terra Média: Sombras de Mordor</p>
        <p>R$&nbsp;79.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Horizon Zero Dawn</p>
        <p>R$&nbsp;115.80</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>The Witcher III Wild Hunt</p>
        <p>R$&nbsp;119.50</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>FIFA 18</p>
        <p>R$&nbsp;195.39</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Super Mario Odyssey</p>
        <p>R$&nbsp;197.88</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Call Of Duty WWII</p>
        <p>R$&nbsp;249.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
</div>
`;

exports[`products.vue render products list ordered by score 1`] = `
<div class="products">
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Terra Média: Sombras de Mordor</p>
        <p>R$&nbsp;79.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Call Of Duty Infinite Warfare</p>
        <p>R$&nbsp;49.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Super Mario Odyssey</p>
        <p>R$&nbsp;197.88</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Mortal Kombat XL</p>
        <p>R$&nbsp;69.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Call Of Duty WWII</p>
        <p>R$&nbsp;249.99</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>The Witcher III Wild Hunt</p>
        <p>R$&nbsp;119.50</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Horizon Zero Dawn</p>
        <p>R$&nbsp;115.80</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>FIFA 18</p>
        <p>R$&nbsp;195.39</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
  <div class="product-wrapper">
    <div class="product">
      <div class="image"><img src="" alt="fifa 18"></div>
      <div class="info">
        <p>Shards of Darkness</p>
        <p>R$&nbsp;71.94</p>
      </div>
      <div class="action"><button>
          adicionar ao carrinho
        </button></div>
    </div>
  </div>
</div>
`;
